File: C++/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int max_area=0;
        int left=0;
        int right=height.size()-1;

        while(left<right){
            int w = right-left;
            int h=min(height[left],height[right]);
            int area=w*h;
            max_area=max(max_area,area);
            if (height[left]<height[right])
                left++;
            else
                right--;
        }
        return max_area;
    }
};

File: Rust/0011-container-with-most-water.rs
impl Solution {
    pub fn max_area(height: Vec<i32>) -> i32 {
        let (mut left, mut right) = (0, height.len() - 1);
        let mut max_area = 0;

        while left < right {
            let (h_left, h_right) = (height[left], height[right]);
            let height = h_left.min(h_right);
            let width = (right - left) as i32;
            let area = height * width;
            max_area = max_area.max(area);

            if h_left < h_right {
                left += 1;
            } else {
                right -= 1;
            }
        }

        max_area
    }
}
